src/dmd/escape.d

```diff
@@ -776,8 +776,7 @@ ByRef:
 
         // If va's lifetime encloses v's, then error
         if (va &&
-            (va.enclosesLifetimeOf(v) && !(v.storage_class & STC.parameter) ||
-             va.storage_class & STC.ref_ ||
+            (va.enclosesLifetimeOf(v) && !(v.storage_class & STC.parameter && v.storage_class & STC.ref_) ||
```

Comment:
`(STC.parameter | STC.ref_)`

Reply:
Both conditions are required, otherwise one can escape a `ref` to a value parameter, see e.g. `fail_compilation/retscope.d`

```d
@safe bar1(ref char* p, char s)
{
    p = &s;
}
```

Reply:
Ok, then ignore me :-)